engine/resources/file.go

```diff
@@ -137,10 +137,25 @@ func (obj *FileRes) isDir() bool {
 // the case where the mode is not specified. The caller should check obj.Mode is
 // not empty.
 func (obj *FileRes) mode() (os.FileMode, error) {
-	m, err := strconv.ParseInt(obj.Mode, 8, 32)
+	if n, err := strconv.ParseInt(obj.Mode, 8, 32); err == nil {
+		return os.FileMode(n), nil
+	}
+
+	// Try parsing symbolic by first getting the files current mode.
+	m := os.FileMode(0)
```

Comment:
You can inline this section and not do this awkward initialization. Eg:

```
stat, err := ...
if err != nil {
...
}
m := stat.Mode()
```

Better?